spec/versioner_spec.rb

```diff
@@ -0,0 +1,174 @@
+    end
+
+    def version(env)
+      @version
+    end
+  end
+
+  let(:routes) { {} }
+  let(:strategies) { [] }
+  let(:env) { {} }
+
+  let(:versioner) do
+    described_class.new(
+      routes: routes,
+      strategies: strategies,
+      env: env
+    )
+  end
+
+  describe '#errors?' do
+    subject { versioner.send(:errors?) }
+    it 'be true' do
+      expect(subject).to be_truthy
+    end
+
+    context 'When no Errors' do
+      let(:strategies) { [Phenotype::HeaderStrategy.new(header: 'API_VERSION')] }
+      let(:routes) { { 1 => '1', 2 => '2' } }
+      let(:env) { { 'HTTP_API_VERSION' => '2' } }
+      subject { versioner.send(:errors?) }
+
+      it 'be false' do
+        expect(subject).to be_falsey
+      end
+    end
+  end
+
+  describe '#maybe_display_errors' do
+    context 'No Strategies' do
+      subject { versioner.send(:display_errors) }
+      it 'return 406' do
+        expect(subject).to eql([406, {'Content-Type' => 'text/html'}, ['No Strategies provided']])
+      end
+
+      it 'has errors' do
+        expect(versioner.send(:errors?)).to be_truthy
+      end
+    end
+
+    context 'To Many Strategies' do
+      let(:strategies) { [MockStrategy.new(1), MockStrategy.new(2)] }
+      subject { versioner.send(:display_errors) }
+
+      it 'return 400' do
+        expect(subject).to(
+          eql([400, {'Content-Type' => 'text/html'}, ['Too mant strategies provided, Supports only one']])
+        )
+      end
+
+      it 'has errors' do
+        expect(versioner.send(:errors?)).to be_truthy
+      end
+    end
+
+    context 'To Many Strategies' do
+      let(:strategies) { [Phenotype::HeaderStrategy.new(header: 'API_VERSION')] }
+      let(:routes) { { 1 => '1', 2 => '2' } }
+      let(:env) { { 'HTTP_API_VERSION' => '3' } }
+
+      subject { versioner.send(:display_errors) }
+
+      it 'return 406' do
+        expect(subject).to(
+          eql([406,
+               { 'Content-Type' => 'text/html' },
+               ["Version: #{versioner.send(:current_version)} is not found"]])
+        )
+      end
+
+      it 'has errors' do
+        expect(versioner.send(:errors?)).to be_truthy
+      end
+    end
+  end
+
+  describe '#cascading_versions' do
+    let(:strategies) { [Phenotype::HeaderStrategy.new(header: 'API_VERSION')] }
+    let(:routes) { { 1 => '1', 2 => '2', 3 => '3' } }
+    subject { versioner.send(:cascading_versions) }
+
+    context 'With all versions' do
+      let(:env) { { 'HTTP_API_VERSION' => '3' } }
+      it 'returns Array of supported versions' do
+        expect(subject).to eql([3, 2, 1])
+      end
+    end
+
+    context 'With subset versions' do
+      let(:env) { { 'HTTP_API_VERSION' => '2' } }
+      it 'returns Array of supported versions' do
+        expect(subject).to eql([2, 1])
+      end
+    end
+  end
+
+  describe '#call' do
+    subject { versioner.call }
+
+    context 'When Errors exist' do
+      it 'returns 406 error' do
+        expect(subject.first).to eql(406)
```

Comment:
do not want to introduce another gem just for this matcher, updated `subject` to include `.first`